### ISSUE TITLE
[PATCH] Fix compiler warning about empty if body

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#12376]
Migrated from [rt.perl.org#114744](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=114744) (status was 'resolved')

Searchable as RT114744$

**Comment by p5pRT** [comment]
### From @jdhedden

#### Created by @jdhedden



Attached patch is meant to correct the following 'blead' build warning&#8203;:

op\.c&#8203;: In function 'Perl\_op\_free'&#8203;:
op\.c&#8203;:713&#8203;:30&#8203;: warning&#8203;: suggest braces around empty body in an 'if' statement

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.16.1:

Configured by Jerry at Tue Aug 14 12:59:06 EDT 2012.

Summary of my perl5 (revision 5 version 16 subversion 1 patch 53802)
configuration:
  Snapshot of: 06d742c02d396d6515581002f376b55ab1972c1b
  Platform:
    osname=cygwin, osvers=1.7.16(0.26253), archname=cygwin-thread-multi-64int
    uname='cygwin_nt-5.1 med-heddenj 1.7.16(0.26253) 2012-07-20 22:55
i686 cygwin '
    config_args='-de -Duse64bitint -Dusethreads -Uusemymalloc
-Dinc_version_list=none -Dnoextensions=DB_File Devel/DProf Devel/Peek
GDBM_File IPC/SysV NDBM_File ODBM_File Sys/Syslog Text/Soundex
Time/Piece attrs B/Debug B/Lint -A append:ccflags= -DNO_MATHOMS -A
define:optimize=-Os -pipe -funit-at-a-time -march=pentium4
-mfpmath=sse -mieee-fp -mmmx -msse -msse2'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__
-DNO_MATHOMS -fno-strict-aliasing -pipe -fstack-protector
-I/usr/local/include',
    optimize='-Os -pipe -funit-at-a-time -march=pentium4 -mfpmath=sse
-mieee-fp -mmmx -msse -msse2',
    cppflags='-DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__ -DNO_MATHOMS
-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.5.3', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='double', nvsize=8,
Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='g++', ldflags =' -Wl,--enable-auto-import
-Wl,--export-all-symbols -Wl,--enable-auto-image-base -s
-fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /lib
    libs=-ldb -ldl -lcrypt
    perllibs=-ldl -lcrypt
    libc=/usr/lib/libc.a, so=dll, useshrplib=true, libperl=cygperl5_16_1.dll
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' -s'
    cccdlflags=' ', lddlflags=' --shared  -Wl,--enable-auto-import
-Wl,--export-all-symbols -Wl,--enable-auto-image-base -s
-L/usr/local/lib -fstack-protector'

Locally applied patches:



@INC for perl 5.16.1:
    /usr/lib/perl5/site_perl/5.16.1/cygwin
    /usr/lib/perl5/site_perl/5.16.1
    /usr/lib/perl5/5.16.1/cygwin
    /usr/lib/perl5/5.16.1
    .


Environment for perl 5.16.1:
    CYGWIN=nodosfilewarning
    HOME=/home/heddenj
    LANG=C
    LANGUAGE=C
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/heddenj/bin:/usr/local/src/perl/bin:/usr/local/bin:/usr/bin:/c/Progra~1/Java/jre6/bin:/c/WINDOWS/system32:/c/WINDOWS:/c/WINDOWS/System32/Wbem:/c/Program
Files/ATI Technologies/ATI.ACE/Core-Static:/c/Program
Files/TortoiseSVN/bin:/c/PYTHON27:/c/Program Files/MySQL/MySQL Server
5.5/bin:/c/WINDOWS/system32/WindowsPowerShell/v1.0:.
    PERLIO=perlio
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @jdhedden

<details><summary>0001-Fix-compiler-warning-about-empty-if-body.patch</summary>

```diff
From f59eed07b124afd2f9733a0f9ff745e39968ca52 Mon Sep 17 00:00:00 2001
From: "Jerry D. Hedden" <jdhedden@cpan.org>
Date: Tue, 4 Sep 2012 13:19:26 -0400
Subject: [PATCH] Fix compiler warning about empty if body

This is meant to correct the following 'blead' build warning:

op.c: In function 'Perl_op_free':
op.c:713:30: warning: suggest braces around empty body in an 'if' statement
---
 op.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/op.c b/op.c
index cdfd7e4..9e176ad 100644
--- a/op.c
+++ b/op.c
@@ -709,8 +709,9 @@ Perl_op_free(pTHX_ OP *o)
     if (type == OP_NULL)
 	type = (OPCODE)o->op_targ;
 
-    if (o->op_slabbed)
-        Slab_to_rw(OpSLAB(o));
+    if (o->op_slabbed) {
+	Slab_to_rw(OpSLAB(o));
+    }
 
     /* COP* is not cleared by op_clear() so that we may track line
      * numbers etc even after null() */
-- 
1.7.9


```
</details>

**Comment by p5pRT** [comment]
### From @nwc10

On Tue\, Sep 04\, 2012 at 10&#8203;:26&#8203;:39AM \-0700\, Jerry D\. Hedden wrote&#8203;:

> \[Please describe your issue here\]
> 
> Attached patch is meant to correct the following 'blead' build warning&#8203;:
> 
> op\.c&#8203;: In function 'Perl\_op\_free'&#8203;:
> op\.c&#8203;:713&#8203;:30&#8203;: warning&#8203;: suggest braces around empty body in an 'if' statement

Thanks for spotting this\. I've pushed this alternative slightly shorter
fix to a smoke\-me branch&#8203;:

commit ca9b5c9aadf1089989ab507b417abaad756b3b0e
Author&#8203;: Nicholas Clark \<nick@&#8203;ccl4\.org>
Date&#8203;:   Tue Sep 4 23&#8203;:05&#8203;:09 2012 \+0200

&nbsp;    Fix a warning about an empty body for Slab\_to\_rw\(\) for the default build\.
&nbsp;    
&nbsp;    Slab\_to\_rw\(\) is only defined as a function with \-DPERL\_DEBUG\_READONLY\_OPS\.
&nbsp;    Warning spotted by Jerry D\. Hedden\.

<details><summary>Inline Patch</summary>

```diff
diff --git a/op.c b/op.c
index cdfd7e4..142ba30 100644
--- a/op.c
+++ b/op.c
@@ -283,7 +283,7 @@ Perl_Slab_to_rw(pTHX_ OPSLAB *const slab)
 }
 
 #else
-#  define Slab_to_rw(op)
+#  define Slab_to_rw(op)    NOOP
 #endif
 
 /* This cannot possibly be right, but it was copied from the old slab



```
</details>
Assuming that doesn't show up anything "obvious" that I'd missed, I'd
prefer to go with this fix\, because it feels more natural \- with it
Slab\_to\_rw\(\) now behaves like a single statement at all times\.

Nicholas Clark

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Sep 04 10&#8203;:26&#8203;:39 2012\, jdhedden@&#8203;cpan\.org wrote&#8203;:
> This is a bug report for perl from jdhedden@&#8203;cpan\.org\,
> generated with the help of perlbug 1\.39 running under perl 5\.16\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> Attached patch is meant to correct the following 'blead' build
> warning&#8203;:
> 
> op\.c&#8203;: In function 'Perl\_op\_free'&#8203;:
> op\.c&#8203;:713&#8203;:30&#8203;: warning&#8203;: suggest braces around empty body in an 'if'
> statement

Thank you\.  Applied as 9bcdb3d\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Sep 04 14&#8203;:13&#8203;:44 2012\, nicholas wrote&#8203;:
> On Tue\, Sep 04\, 2012 at 10&#8203;:26&#8203;:39AM \-0700\, Jerry D\. Hedden wrote&#8203;:
> 
> > \[Please describe your issue here\]
> >
> > Attached patch is meant to correct the following 'blead' build
> warning&#8203;:
> >
> > op\.c&#8203;: In function 'Perl\_op\_free'&#8203;:
> > op\.c&#8203;:713&#8203;:30&#8203;: warning&#8203;: suggest braces around empty body in an 'if'
> statement
> 
> Thanks for spotting this\. I've pushed this alternative slightly
> shorter
> fix to a smoke\-me branch&#8203;:
> 
> commit ca9b5c9aadf1089989ab507b417abaad756b3b0e
> Author&#8203;: Nicholas Clark \<nick@&#8203;ccl4\.org>
> Date&#8203;:   Tue Sep 4 23&#8203;:05&#8203;:09 2012 \+0200
> 
>     Fix a warning about an empty body for Slab\_to\_rw\(\) for the default
> build\.
> 
>     Slab\_to\_rw\(\) is only defined as a function with
> \-DPERL\_DEBUG\_READONLY\_OPS\.
>     Warning spotted by Jerry D\. Hedden\.
> 
> diff \-\-git a/op\.c b/op\.c
> index cdfd7e4\.\.142ba30 100644
> \-\-\- a/op\.c
> \+\+\+ b/op\.c
> @&#8203;@&#8203; \-283\,7 \+283\,7 @&#8203;@&#8203; Perl\_Slab\_to\_rw\(pTHX\_ OPSLAB \*const slab\)
>  \}
> 
>  \#else
> \-\#  define Slab\_to\_rw\(op\)
> \+\#  define Slab\_to\_rw\(op\)    NOOP
>  \#endif
> 
>  /\* This cannot possibly be right\, but it was copied from the old slab
> 
> 
> Assuming that doesn't show up anything "obvious" that I'd missed\, I'd
> prefer to go with this fix\, because it feels more natural \- with it
> Slab\_to\_rw\(\) now behaves like a single statement at all times\.

Sorry\, I missed that\.  Feel free to revert 9bcdb3d\.


\-\- 

Father Chrysostomos
